types/rdf-js/index.d.ts

```diff
@@ -476,3 +476,181 @@ export interface DatasetCoreFactory<Q extends BaseQuad = Quad> {
      */
     dataset(quads?: Q[]): DatasetCore<Q>;
 }
+
+export interface Dataset<Q extends BaseQuad = Quad> extends DatasetCore<Q> {
+    match(subject?: Term, predicate?: Term, object?: Term, graph?: Term): Dataset<Q>;
+
+    /**
+     * Imports the quads into this dataset.
+     *
+     * This method differs from `Dataset.union` in that it adds all `quads` to the current instance, rather than
+     * combining `quads` and the current instance to create a new instance.
+     */
+    addAll(quads: Dataset<Q>|Q[]): this;
+
+    /**
+     * Returns `true` if the current instance is a superset of the given dataset; differently put: if the given dataset
+     * is a subset of, is contained in the current dataset.
+     *
+     * Blank Nodes will be normalized.
+     */
+    contains(other: Dataset<Q>): boolean;
+
+    /**
+     * This method removes the quads in the current instance that match the given arguments.
+     *
+     * The logic described in {@link https://rdf.js.org/dataset-spec/#quad-matching|Quad Matching} is applied for each
+     * quad in this dataset to select the quads which will be deleted.
+     *
+     * @param subject   The optional exact subject to match.
+     * @param predicate The optional exact predicate to match.
+     * @param object    The optional exact object to match.
+     * @param graph     The optional exact graph to match.
+     */
+    deleteMatches(subject?: Term, predicate?: Term, object?: Term, graph?: Term): this;
```

Comment:
Will be updated if https://github.com/DefinitelyTyped/DefinitelyTyped/pull/40821 is approved.